src/components/PasswordField/PasswordField.vue

```diff
@@ -0,0 +1,322 @@
+		 */
+		success: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * Toggles the error state of the component. Adds an error icon.
+		 * this cannot be used together with canClear.
+		 */
+		error: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * Additional error message
+		 *
+		 * This will be displayed beneath the input field
+		 */
+		helperText: {
+			type: String,
+			default: '',
+		},
+
+		/**
+		 * The minlength property defines the minimum number of characters
+		 * (as UTF-16 code units) the user can enter
+		 */
+		minlength: {
+			type: Number,
+			default: 0,
+		},
+
+		/**
+		 * The maxlength property defines the maximum number of characters
+		 * (as UTF-16 code units) the user can enter
+		 */
+		maxlength: {
+			type: Number,
+			default: null,
+		},
+
+		/**
+		 * Helps the browser identify the type of password field and to provide
+		 * autocompletion for the password. Allowed values are off, on, new-password
+		 * old-password
+		 *
+		 * By default try to autocomplete with the current password
+		 */
+		autocomplete: {
+			type: String,
+			validator: (value) => [
+				'new-password',
+				'current-password',
+				'one-time-code',
+				'on',
+				'off',
+			].includes(value),
+			default: 'current-password',
+		},
+
+		/**
+		 * Check if the user entered a valid password using the password_policy
+		 * app if available.
+		 *
+		 * Warning: this doesn't replace server side checking and will do nothing
+		 * if the password_policy app is disabled.
+		 */
+		checkPasswordStrength: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * Id of the input field. To use when using an external label
+		 */
+		id: {
+			type: String,
+			default: '',
+		},
+	},
+
+	data() {
+		return {
+			isPasswordHidden: true,
+			internalHelpMessage: '',
+			passwordPolicy: defaultPasswordPolicy,
+		}
+	},
+
+	computed: {
+		computedHelperText() {
+			if (this.helperText.length > 0) {
+				return this.helperText
+			}
+			return this.internalHelpMessage
+		},
+		rules() {
+			const { minlength, passwordPolicy } = this
+			return {
+				minlength: minlength ?? passwordPolicy?.minLength,
+			}
+		},
+		trailingButtonLabel() {
+			return this.isPasswordHidden ? t('Show password') : t('Hide password')
+		},
+	},
+
+	watch: {
+		value(newValue) {
+			if (this.checkPasswordStrength) {
+				if (this.passwordPolicy === null) {
+					return
+				}
+				if (this.passwordPolicy) {
+					this.checkPassword(newValue)
+				}
+			}
+		},
+	},
+
+	methods: {
+		handleInput(event) {
+			this.$emit('update:value', event.target.value)
+		},
+		togglePasswordVisibility() {
+			this.isPasswordHidden = !this.isPasswordHidden
+		},
+		checkPassword: debounce(async function(password) {
+			const { data } = await axios.post(generateOcsUrl('apps/password_policy/api/v1/validate'), { password })
+			if (data.ocs.data.passed) {
+				this.internalHelpMessage = t('Password is secure')
+			}
+			this.internalHelpMessage = data.reason
```

Comment:
What is the expected behavior when this request fail?

Reply:
The expected behavior is that it doesn't fail. I added a try-catch and a logger to log just in case.